### ISSUE TITLE
Integration with Wolfram Alpha API for Calculations

### ISSUE THREAD

**Issue by delizin** [LAION-AI/Open-Assistant#595]
I'd like to share this article from Stephen Wolfram, of Wolfram Alpha:
https://writings.stephenwolfram.com/2023/01/wolframalpha-as-the-way-to-bring-computational-knowledge-superpowers-to-chatgpt/

Connecting Open-Assistant to the Wolfram|Alpha API could provide a powerful solution for many problems that require high precision. Open-Assistant will be able to produce text that is similar to human-generated text, making it useful for tasks like writing suggestions and text generation. However, it may end up like ChatGPT where it does not have a precise understanding of the information it generates and can fall short when asked for precise information.

On the other hand, Wolfram|Alpha is a knowledge-based system that has a precise computational representation of knowledge, which allows it to compute things reliably. By connecting Open-Assistant with Wolfram|Alpha, we can overcome these limitations, as it can produce natural language, which Wolfram|Alpha can then convert into a precise, formal representation for reliable computation.

This combination provides a powerful solution that can be applied to a wide range of tasks.

**Comment by GravermanDev** [comment]
This is cool however it's impossible due because of our license and the fact that the API is paid. But actually there is a wolfram like open source alternative called Sage. It could work

**Comment by GravermanDev** [comment]
nvm Sage is sometimes not commercial, well we gotta look further, like sympy gamma

**Comment by yk** [comment]
I mean, it's totally possible for us to build in support for WA. Like, it could be an optional extension, and you'd have to plug in your own API key, but just because we're Apache doesn't mean we can't call a paid service if the user wants that to happen.

**Comment by danielpatrickhug** [comment]
I spotted this in langchain's issues thread. Stephen Wolfram talking about integrating wolfram and chatgpt. He seems pretty optimistic. https://writings.stephenwolfram.com/2023/01/wolframalpha-as-the-way-to-bring-computational-knowledge-superpowers-to-chatgpt/

**Comment by sedthh** [comment]
I think the Assistant should allow connection to multiple 3rd party APIs as long as the user has their own key.
Weather, news, stocks, etc. should all be retrievable just like with any other smart assitant that has access to the internet. 

I think the best way to achieve this would be to introduce an API wrapper to the logic that turns any 3rd party JSON into a single, standardized format so in case of an API change or switch of vendor, the funcitonality will not break.

I can work on this and other APIs as well if others agree.

**Comment by MalikMAlna** [comment]
Yeah @sedthh, I think that would be helpful, but would likely need to be made into its own board task.

**Comment by mm12** [comment]
Remember there are also alternatives to Wolfram, might not be as good tho.

**Comment by davidak** [comment]
As you probably heard, ChatGPT does now have Plugin Support including WolframAlpha.

https://openai.com/blog/chatgpt-plugins

It should be preferred to do everything locally (offline) with open source software and only use online services (APIs) when needed and also prefer community maintained open data sources. Proprietary paid services should be the last thing to try.

**Comment by LuposX** [comment]
[Toolformer Paper](https://arxiv.org/abs/2302.04761) might be helpful.

**Comment by pevogam** [comment]
I would relate this issue to this pull request as a specific approach: https://github.com/LAION-AI/Open-Assistant/pull/2765
